pkg/eventpersister/persister/persister.go

```diff
@@ -535,14 +535,14 @@ func getVariationID(reason featureproto.Reason_Type, vID string) string {
 func (p *Persister) upsertEvaluationCount(event proto.Message, environmentNamespace string) error {
 	if e, ok := event.(*eventproto.EvaluationEvent); ok {
 		vID := getVariationID(e.Reason.Type, e.VariationId)
-		eck := p.newEvaluationCountkey(eventCountKey, e.FeatureId, vID, environmentNamespace, e.Timestamp)
-		if err := p.countEvent(eck); err != nil {
-			return err
-		}
 		uck := p.newEvaluationCountkey(userCountKey, e.FeatureId, vID, environmentNamespace, e.Timestamp)
 		if err := p.countUser(uck, e.UserId); err != nil {
 			return err
 		}
+		eck := p.newEvaluationCountkey(eventCountKey, e.FeatureId, vID, environmentNamespace, e.Timestamp)
```

Comment:
I changed the order of EventCount and UserCount because if the EventCount succeeded but the UserCount failed, it will increment one more to EventCount until the UserCount succeeds.
The UserCount won't change if the same user sends the same event because it is unique.